src/creature_states_combt.c

```diff
@@ -3199,7 +3199,7 @@ TbBool creature_look_for_enemy_object_combat(struct Thing* thing)
         struct CreatureControl* cctrl = creature_control_get_from_thing(thing);
         objtng = thing_get(cctrl->combat.battle_enemy_idx);
         if (thing_is_dungeon_heart(objtng)) {
-            return false;
+            return true;
```

Comment:
Would you please enlighten me about why you need to modify this line and line 3127?
Isn't the modification to process_creature_self_spell_casting() enough?